components/ai_chat/core/browser/model_validator.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ai_chat/core/browser/model_validator.h"
+
+#include <limits>
+#include <optional>
+
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-forward.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-shared.h"
+#include "brave/components/ai_chat/core/common/mojom/settings_helper.mojom.h"
+#include "brave/net/base/url_util.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+
+namespace ai_chat {
+
+// Static
+bool ModelValidator::IsValidContextSize(const std::optional<int32_t>& size) {
+  if (!size.has_value()) {
+    return false;
+  }
+
+  base::CheckedNumeric<size_t> checked_value(size.value());
+
+  return checked_value.IsValid() &&
+         checked_value.ValueOrDie() >= kMinCustomModelContextSize &&
+         checked_value.ValueOrDie() <= kMaxCustomModelContextSize;
+}
+
+// Static
+bool ModelValidator::HasValidContextSize(const mojom::Model& model) {
+  if (model.options->is_custom_model_options()) {
+    return IsValidContextSize(
+        model.options->get_custom_model_options()->context_size);
+  }
+
+  // Should not be reached
+  DCHECK(false);
```

Comment:
This not reached situation is problematic, the interface won't be able to restrict custom or builtin models to be passed. When it is not custom model, we can assume the context size is always valid and please add a test case for that. 
If ModelValidator should only be used by custom model, then the input should be `mojom::CustomModelOptions` instead of `mojom::Model`